reviews/views.py

```diff
@@ -1,3 +1,62 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http      import JsonResponse, request
+from django.views     import View
+from reviews.models   import Review
+from users.models     import User
+from products.models  import Product
+from users.utils      import login_decorator
+
+class ReviewView(View):
+    def get(self,request, review_id):
+        reviews = Review.objects.all()
+        result = []
+
+        if not Review.objects.filter(id=review_id).exists():
+            return JsonResponse ({"MESSAGE": "REVIEW_DOES_NOT_EXISTS"}, status = 400)   
+ 
+        result = [
+            {
+                "user"            : review.user.id,
+                "product_id"      : review.product_id ,
+                "create_at"       : review.created_at,
+                "size_rating"     : review.size_rating,
+                "color_rating"    : review.color_rating,
+                "delivery_rating" : review.delivery_rating,
+                "title"           : review.title,
+                "comment"         : review.comment,
+
+            } for review in reviews]
+
+        return JsonResponse ({"result": result}, status = 200)
+
+    @login_decorator
+    def delete (self,request,review_id):
+
+        if not Review.objects.filter(id=review_id).exists():
+            return JsonResponse({"MESSAGE": "NO_REVIEWS"}, status = 400)
+
+        if Review.objects.get(id=review_id) != request.user :
+            return JsonResponse({"MESSAGE": "UNAUTHROIZED"}, status = 400)
+
+        Review.objects.filter(id=review_id).delete()
+
+        return JsonResponse ({"MESSAGE" : "DELETE"}, status = 204)    
+
+    @login_decorator
+    def patch (self,request, review_id):
+        try:
+            data = json.loads(request.body)
+
+            if not Review.objects.filter(id=review_id).exists():
+                return JsonResponse({"MESSAGE": "NO_REVIEWS"}, status = 400)
+
+            if Review.objects.get(id=review_id) != request.user :
+                return JsonResponse({"MESSAGE": "UNAUTHROIZED"}, status = 400)
```

Comment:
```suggestion
            if not Review.objects.filter(id=review_id, user=request.user).exists():
                return JsonResponse({"MESSAGE": "NO_REVIEWS"}, status = 400)
```
이런 식으로 조건을 한 번에 걸어서 판단하는 것이 좋습니다. 2번째에 대한 if를 안 해도 되고 가독성도 좋아집니다.

Reply:
넵 수정했습니다 :-)